state/machine.go

```diff
@@ -459,29 +459,37 @@ func (m *Machine) IsManager() bool {
 
 // IsManual returns true if the machine was manually provisioned.
 func (m *Machine) IsManual() (bool, error) {
+	// To avoid unnecessary db lookups, a little of the
+	// logic from isManualMachine() below is duplicated here
+	// so we can exit early if possible.
+	if strings.HasPrefix(m.doc.Nonce, manualMachinePrefix) {
+		return true, nil
+	}
+	if m.doc.Id != "0" {
+		return false, nil
+	}
+	model, err := m.st.Model()
```

Comment:
Do we really want to load the model just to get the config?

Reply:
Outside the state package, no choice, but since we're in state, we can read the setting collection directly, so I made that change